stripe/resource.py

```diff
@@ -357,10 +357,13 @@ def class_url(cls):
 
     def instance_url(self):
         id = self.get('id')
-        if not id:
+
+        if not isinstance(id, (str, unicode)):
```

Comment:
`isinstance(id, basestring)` would be slightly more Pythonic :)

Reply:
See [#7ae76e8](https://github.com/stripe/stripe-python/pull/339/commits/7ae76e8e4fe549b2e0f0c8a7bd89ad7f83678362) as for why I chose `(str, unicode)`.

In summary, it was a Python 2 compatibility issue concerning the following behavior:

```
Python 2.7.13 (default, Jul 21 2017, 03:24:34) 
[GCC 7.1.1 20170630] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> str == unicode
False
```

So `if not isinstance(id, str)` was failing with Python 2.

Reply:
[`basestring`](https://docs.python.org/2/library/functions.html#basestring) is the abstract parent class for both `str` and `unicode` in Python 2, and is translated to `str` by `2to3`.

Reply:
Okay, I will use `basestring` as I didn't completely realize stripe-python was using `2to3`. Thank you for the suggestion!